Tests/SonarScanner.MSBuild.PreProcessor.Test/PreprocessorObjectFactoryTests.cs

```diff
@@ -29,77 +31,107 @@ namespace SonarScanner.MSBuild.PreProcessor.Test
+        }
 
-        #region Private methods
+        [TestMethod]
+        public void CreateHttpClient_UserAgent()
+        {
+            var userAgent = GetHeader(PreprocessorObjectFactory.CreateHttpClient(null, null, null, null), HttpRequestHeader.UserAgent);
+
+            var scannerVersion = typeof(WebClientDownloaderTest).Assembly.GetName().Version.ToDisplayString();
+            userAgent.Should().Be($"ScannerMSBuild/{scannerVersion}");
+        }
 
-        private ProcessedArgs CreateValidArguments()
+        [TestMethod]
+        public void CreateHttpClient_SemicolonInUsername()
         {
-            var logger = new TestLogger();
-            var cmdLineArgs = new Common.ListPropertiesProvider();
-            cmdLineArgs.AddProperty(Common.SonarProperties.HostUrl, "http://foo");
-
-            var validArgs = new ProcessedArgs("key", "name", "verions", "organization", false,
-                cmdLineArgs,
-                new Common.ListPropertiesProvider(),
-                EmptyPropertyProvider.Instance,
-                logger);
-            return validArgs;
+            Action act = () => PreprocessorObjectFactory.CreateHttpClient("user:name", null, null, null);
+            act.Should().ThrowExactly<ArgumentException>().WithMessage("username cannot contain the ':' character due to basic authentication limitations");
+        }
+
+        [TestMethod]
+        public void CreateHttpClient_AccentsInUsername()
+        {
+            Action act = () => PreprocessorObjectFactory.CreateHttpClient("héhé", "password", null, null);
+            act.Should().ThrowExactly<ArgumentException>().WithMessage("username and password should contain only ASCII characters due to basic authentication limitations");
+        }
+
+        [TestMethod]
+        public void CreateHttpClient_AccentsInPassword()
+        {
+            Action act = () => PreprocessorObjectFactory.CreateHttpClient("username", "héhé", null, null);
+            act.Should().ThrowExactly<ArgumentException>().WithMessage("username and password should contain only ASCII characters due to basic authentication limitations");
         }
 
-        #endregion Private methods
+        [TestMethod]
+        public void CreateHttpClient_UsingClientCert()
+        {
+            Action act = () => PreprocessorObjectFactory.CreateHttpClient(null, null, "certtestsonar.pem", "dummypw");
+            act.Should().NotThrow();
+        }
+
+        private static string GetHeader(HttpClient client, HttpRequestHeader header) =>
+            client.DefaultRequestHeaders.Contains(header.ToString())
+                ? string.Join(";", client.DefaultRequestHeaders.GetValues(header.ToString()))
+                : null;
+
+        private static ProcessedArgs CreateValidArguments()
+        {
+            var logger = new TestLogger();
+            var cmdLineArgs = new ListPropertiesProvider();
+            cmdLineArgs.AddProperty(SonarProperties.HostUrl, "https://sonarsource.com");
+
```

Comment:
Can be single-lined.

The `Property` class desperately needs a constructor that takes 2 arguments (not in this PR)
```suggestion
            var cmdLineArgs = new ListPropertiesProvider(new[] { new Property() { Id = SonarProperties.HostUrl, Value = "https://sonarsource.com" } });
```